client-java/controller-api/src/main/java/org/evomaster/client/java/controller/api/dto/SutRunDto.java

```diff
@@ -18,12 +18,27 @@
      */
     public Boolean calculateSqlHeuristics;
 
+    /**
+     *  Whether SQL execution info should be saved.
+     */
+    public Boolean extractSqlExecutionInfo;
+
     public SutRunDto() {
     }
 
+    public SutRunDto(Boolean run, Boolean resetState, Boolean calculateSqlHeuristics, Boolean extractSqlExecutionInfo) {
+        if (calculateSqlHeuristics != null && calculateSqlHeuristics && extractSqlExecutionInfo != null && !extractSqlExecutionInfo)
+            throw new IllegalArgumentException("extractSqlExecutionInfo should be enabled when calculateSqlHeuristics is enabled");
+        this.run = run;
+        this.resetState = resetState;
+        this.calculateSqlHeuristics = calculateSqlHeuristics;
+        this.extractSqlExecutionInfo = extractSqlExecutionInfo;
+    }
+
     public SutRunDto(Boolean run, Boolean resetState, Boolean calculateSqlHeuristics) {
         this.run = run;
         this.resetState = resetState;
         this.calculateSqlHeuristics = calculateSqlHeuristics;
+        this.extractSqlExecutionInfo = calculateSqlHeuristics!=null && calculateSqlHeuristics;
     }
```

Comment:
must avoid copy&paste of same code
this constructor could be rewritten with something like:
```
this(run, resetState, calculateSqlHeuristics, calculateSqlHeuristics )
```

Reply:
fixed, thanks